### PR TITLE
jit: support global get and set instructions

### PR DESCRIPTION
**Pull request by mathetake**
part of #65

### CODE REVIEW

File: wasm/jit/jit_amd64_test.go [hunk 1]
```diff
@@ -1682,3 +1682,93 @@ func TestAmd64Builder_handleSwap(t *testing.T) {
...
+			eng.currentGlobalSliceAddress = uintptr(unsafe.Pointer(&globals[0]))
+			mem := newMemoryInst()
+			jitcall(
+				uintptr(unsafe.Pointer(&code[0])),
+				uintptr(unsafe.Pointer(eng)),
+				uintptr(unsafe.Pointer(&mem.Buffer[0])),
+			)
+			// Check the stack.
+			require.Equal(t, uint64(1), eng.currentStackPointer)
+			require.Equal(t, globals[op.Index].Val, eng.stack[0])
```
Comment (codefromthecrypt) [inline wasm/jit/jit_amd64_test.go:1733]: I'm guessing this means 12345 is at the top of the stack? if true, good comment right? :D

File: wasm/jit/jit_amd64_test.go [hunk 2]
```diff
@@ -1682,3 +1682,93 @@ func TestAmd64Builder_handleSwap(t *testing.T) {
...
+			require.NoError(t, err)
+			builder.returnFunction()
+
+			// Assemble.
+			code, err := builder.assemble()
+			require.NoError(t, err)
+			// Run code.
+			eng := newEngine()
+			eng.currentGlobalSliceAddress = uintptr(unsafe.Pointer(&globals[0]))
+			eng.push(12345)
```
Comment (codefromthecrypt) [inline wasm/jit/jit_amd64_test.go:1762]: similar feedback on this test as the one above. extract a variable like "valueToSet" with 12345 as it teaches more than scanning up and down what's happening

File: wasm/jit/jit_amd64_test.go [hunk 3]
```diff
@@ -1682,3 +1682,93 @@ func TestAmd64Builder_handleSwap(t *testing.T) {
...
+
+func TestAmd64Builder_handleGlobalSet(t *testing.T) {
+	for i, tp := range []wasm.ValueType{
+		wasm.ValueTypeF32, wasm.ValueTypeF64, wasm.ValueTypeI32, wasm.ValueTypeI64,
+	} {
+		tp := tp
+		t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
+			// Setup the globals.
+			builder := requireNewBuilder(t)
+			globals := []*wasm.GlobalInstance{nil, {Val: 0, Type: &wasm.GlobalType{ValType: tp}}, nil}
```
Comment (codefromthecrypt) [inline wasm/jit/jit_amd64_test.go:1746]: guessing 0 is the old value? maybe use something besides zero as it could be conflated with unset.

**Review by codefromthecrypt — APPROVED** [review]
Lots of pedantic things in the nit category. Don't let the fury of these suggest this change is anything but a great job!

File: wasm/jit/engine.go [hunk 4]
```diff
@@ -332,6 +342,7 @@ func (e *engine) exec(f *compiledWasmFunction) {
 		caller:                   nil,
 		continuationStackPointer: f.inputs,
 	}
+	e.currentGlobalSliceAddress = f.globalSliceAddress
```
[the remark this answers is not in the document]
Comment (mathetake) [reply wasm/jit/engine.go:345]: without initializing it here, we might end up accessing invalid memory region as we use this field in the jitted code, and the following for loop starts with entering `jitcall` so we have to initialize here!

File: wasm/jit/jit_amd64_test.go (hunk 3 above)
Comment (mathetake) [reply wasm/jit/jit_amd64_test.go]: https://github.com/tetratelabs/wazero/pull/81/commits/41c8b727346b3703ef164c102dbebf85e68d8c56

File: wasm/jit/jit_amd64_test.go (hunk 2 above)
Comment (mathetake) [reply wasm/jit/jit_amd64_test.go]: https://github.com/tetratelabs/wazero/pull/81/commits/41c8b727346b3703ef164c102dbebf85e68d8c56

File: wasm/jit/jit_amd64_test.go (hunk 1 above)
Comment (mathetake) [reply wasm/jit/jit_amd64_test.go]: https://github.com/tetratelabs/wazero/pull/81/commits/41c8b727346b3703ef164c102dbebf85e68d8c56

File: wasm/jit/jit_amd64_test.go [hunk 5]
```diff
@@ -1682,3 +1682,93 @@ func TestAmd64Builder_handleSwap(t *testing.T) {
...
+
+func TestAmd64Builder_handleGlobalGet(t *testing.T) {
+	for i, tp := range []wasm.ValueType{
+		wasm.ValueTypeF32, wasm.ValueTypeF64, wasm.ValueTypeI32, wasm.ValueTypeI64,
+	} {
+		tp := tp
+		t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
+			// Setup the globals.
+			builder := requireNewBuilder(t)
+			globals := []*wasm.GlobalInstance{nil, {Val: 12345, Type: &wasm.GlobalType{ValType: tp}}, nil}
```
[the remark this answers is not in the document]
Comment (mathetake) [reply wasm/jit/jit_amd64_test.go]: https://github.com/tetratelabs/wazero/pull/81/commits/41c8b727346b3703ef164c102dbebf85e68d8c56

File: wasm/jit/jit_amd64_test.go [hunk 6]
```diff
@@ -1682,3 +1682,93 @@ func TestAmd64Builder_handleSwap(t *testing.T) {
...
+			// Run code.
+			eng := newEngine()
+			eng.currentGlobalSliceAddress = uintptr(unsafe.Pointer(&globals[0]))
+			mem := newMemoryInst()
+			jitcall(
+				uintptr(unsafe.Pointer(&code[0])),
+				uintptr(unsafe.Pointer(eng)),
+				uintptr(unsafe.Pointer(&mem.Buffer[0])),
+			)
+			// Check the stack.
```
[the remark this answers is not in the document]
Comment (mathetake) [reply wasm/jit/jit_amd64_test.go]: https://github.com/tetratelabs/wazero/pull/81/commits/0e950823722eebf9a3ae2b30d8d6630bc6a81222

**Comment by mathetake** [comment]
@codefromthecrypt thanks for the detailed reviews! Improved a lot :D

**Pull request merged**